src/realm/query_value.cpp

```diff
@@ -55,13 +55,19 @@ static const std::vector<std::pair<std::string, TypeOfValue::Attribute>> attribu
     {"double", TypeOfValue::Double},
     {"decimal128", TypeOfValue::Decimal128},
     {"decimal", TypeOfValue::Decimal128},
-    {"object", TypeOfValue::ObjectLink},
+    {"objectlink", TypeOfValue::ObjectLink},
     {"link", TypeOfValue::ObjectLink},
     {"objectid", TypeOfValue::ObjectId},
     {"uuid", TypeOfValue::UUID},
     {"guid", TypeOfValue::UUID},
     {"numeric", TypeOfValue::Numeric},
-    {"bindata", TypeOfValue::Binary}};
+    {"bindata", TypeOfValue::Binary},
+    {"object", TypeOfValue::Object},
+    {"dictionary", TypeOfValue::Object},
```

Comment:
When doing the query string in the JS SDK, this will likely be ambiguous since our `"object"` data type means "Realm object".